fvm/errors/errors.go

```diff
@@ -114,6 +114,39 @@ func HandleRuntimeError(err error) error {
 		return NewUnknownFailure(err)
 	}
 
+	errs := []error{runErr}
+	var parentErr errors.ParentError
+	if As(err, &parentErr) {
+		errs = parentErr.ChildErrors()
+	}
+
+	codedErrs := make([]CodedError, 0)
+	for _, err := range errs {
+		coded, isUnknown := findImportantCodedError(err)
+		if isUnknown {
+			// ignore uncoded errors
+			continue
+		}
+
+		if coded.Code().IsFailure() {
+			return WrapCodedError(
```

Comment:
note: this logic could be simplified a bit if we are OK just returning the code from the first `CodedError` in the `ChildErrors()` list, instead of specifically searching for failures.